### PR TITLE
Final version for thomaswue

### PR DESCRIPTION
**Pull request by thomaswue**
#### Check List:

- [x] You have run `./mvnw verify` and the project builds successfully
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [X] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time: 0.39
* Execution time of reference implementation: 120.37

Added detailed comments for the used native image flags, some code clean up for better naming, some minor tuning. This translates to around 2% faster on my machine, but let's see if this also holds up for the reference machine.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Nice!

```
Benchmark 1: timeout -v 300 ./calculate_average_thomaswue.sh 2>&1
  Time (mean ± σ):      1.830 s ±  0.011 s    [User: 0.002 s, System: 0.004 s]
  Range (min … max):    1.798 s …  1.836 s    10 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Summary
  thomaswue: trimmed mean 1.832794623615, raw times 1.7984286922400001,1.83378488024,1.8358817772400002,1.8351058012400001,1.8321136482400002,1.83353322124,1.83313856524,1.8319531362400001,1.83227779624,1.83044994024

Leaderboard

| # | Result (m:s.ms) | Implementation     | JDK | Submitter     | Notes     |
|---|-----------------|--------------------|-----|---------------|-----------|
|   | 00:01.832 | [link](https://github.com/gunnarmorling/1brc/blob/main/src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java)| 21.0.2-graal | [Thomas Wuerthinger](https://github.com/thomaswue) | GraalVM native binary, uses Unsafe |
```

File: src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java [hunk 1]
```diff
@@ -103,61 +101,123 @@ private static TreeMap<String, Result> accumulateResults(List<Result>[] allResul
...
+            long segmentStart;
+            if (current == fileStart) {
+                segmentStart = current;
+            }
+            else {
+                segmentStart = nextNewLine(current) + 1;
+            }
+
+            long dist = (segmentEnd - segmentStart) / 3;
+            long midPoint1 = nextNewLine(segmentStart + dist);
```
[the remark this answers is not in the document]
Comment (thomaswue) [reply src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java:122]: This way the x86 processor is able to better use its processing units to execute more instructions per cycle. I am preparing some more information and blog post about this.

[the remark this answers is not in the document]
Comment (Alex-Cheng) [reply src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java:122]: Thank you. I am looking forward your blog. I didn't find it in the resource list. Is it in progress? I don't want to miss the great blog.

[the remark this answers is not in the document]
Comment (thomaswue) [reply src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java:122]: It took a while, but a repository with the step-by-step explanation on how the final solution was created and some in-depths explanations are available here: https://github.com/thomaswue/1brc-steps

**Pull request merged**